extensions/mssql/src/sparkFeature/dialog/dialogCommands.ts

```diff
@@ -0,0 +1,144 @@
+import * as vscode from 'vscode';
+const localize = nls.loadMessageBundle();
+
+import { ICommandViewContext, Command, ICommandObjectExplorerContext, ICommandUnknownContext } from '../../objectExplorerNodeProvider/command';
+import { SparkJobSubmissionDialog } from './sparkJobSubmission/sparkJobSubmissionDialog';
+import { AppContext } from '../../appContext';
+import { getErrorMessage } from '../../utils';
+import * as constants from '../../constants';
+import { HdfsFileSourceNode } from '../../objectExplorerNodeProvider/hdfsProvider';
+import { getNode } from '../../objectExplorerNodeProvider/hdfsCommands';
+import * as LocalizedConstants from '../../localizedConstants';
+import { SqlClusterLookUp } from '../../sqlClusterLookUp';
+import { SqlClusterConnection } from '../../objectExplorerNodeProvider/connection';
+
+export class OpenSparkJobSubmissionDialogCommand extends Command {
+	constructor(appContext: AppContext, private outputChannel: vscode.OutputChannel) {
+		super(constants.livySubmitSparkJobCommand, appContext);
+	}
+
+	protected async preExecute(sqlContext: ICommandUnknownContext | ICommandObjectExplorerContext, args: object = {}): Promise<any> {
+		return this.execute(sqlContext, args);
+	}
+
+	async execute(sqlContext: ICommandUnknownContext | ICommandObjectExplorerContext, ...args: any[]): Promise<void> {
+		try {
+			let sqlClusterConnection: SqlClusterConnection = undefined;
+			if (sqlContext.type === constants.ObjectExplorerService) {
+				sqlClusterConnection = SqlClusterLookUp.findSqlClusterConnection(sqlContext, this.appContext);
+			}
+
+			let clusterConnInfo: sqlops.connection.Connection = undefined;
+			if (sqlClusterConnection) {
+				clusterConnInfo = sqlClusterConnection.sqlClusterConnObj;
+			}
+			if (!clusterConnInfo) {
+				let selectedConn = await this.selectConnection();
+				clusterConnInfo = await SqlClusterLookUp.getSqlClusterConnInfo(selectedConn);
+			}
+
+			let dialog = new SparkJobSubmissionDialog(clusterConnInfo, this.appContext, this.outputChannel);
+			await dialog.openDialog();
+		} catch (error) {
+			this.appContext.apiWrapper.showErrorMessage(getErrorMessage(error));
+		}
+	}
+
+	private async selectConnection(): Promise<sqlops.connection.Connection> {
+		let connectionList: sqlops.connection.Connection[] = await this.apiWrapper.getActiveConnections();
+		let displayList: string[] = new Array();
+		let connectionMap: Map<string, sqlops.connection.Connection> = new Map();
+		if (connectionList && connectionList.length > 0) {
+			connectionList.forEach(conn => {
+				if (conn.providerName === constants.sqlProviderName) {
+					displayList.push(conn.options.host);
+					connectionMap.set(conn.options.host, conn);
+				}
+			});
+		}
+
+		if (displayList.length === 0) {
+			throw new Error(localize('sparkJobSubmission_NoActiveHadoopConnection', 'Spark Job submission failed: No active Hadoop connection. Please click the cluster node in the left tree to activate. '));
```

Comment:
Please add tracking issue - should be able to prompt for connection at this point. We added the APIs after this code was first checked in.

Reply:
Do you mean you want me to remove this?